### PR TITLE
docs: add agent-ready specifications and exploratory UI testing standards

### PR DESCRIPTION
**Pull request by cormoran**
## Summary

- Add a Japanese `docs/spec` sitemap, 14 page/feature specifications and 6 shared-module contracts, traced to implementation symbols and tests.
- Standardize requirements versus current behavior, observable flows, persistence, invariants, errors, accepted-defect evidence and unknowns. Link same-PR maintenance rules from AGENTS.md and Copilot instructions; leave the PR template unchanged.
- Add an exploratory browser testing guide and `npm run spec:check` to CI (local links, required sections, duplicate IDs and Page-source coverage).
- Add a thin repository `exploratory-ui-test` skill: change-focused versus whole-app coverage, luna-low exploration, selective terra-high source review, and bounded proactive regression checks. Keep the main testing guide tool-neutral; Orca-specific execution and recovery live in a conditional browser-tools guide.
- Iterate the keymap pilot with luna-low, author remaining scopes with 3 terra-high workers, and validate them with 3 luna-low workers plus bounded follow-ups. Integrate reusable findings into specifications/guides instead of accumulating trial reports in HEAD. Future raw logs belong in temporary output, with summaries in PRs/completion reports.

## Validation

- [x] `npm run spec:check` — 26 documents / 299 specification IDs
- [x] Repository skill frontmatter validation and guide-link checks (no paid browser workers launched for these documentation-only changes)
- [x] `npm run lint`
- [x] `npm test -- --runInBand` — 85 suites, 707 passed, 1 skipped
- [x] `npm run build` — existing large-chunk warning only
- [x] Prettier for changed documents / script, `git diff --check`
- [x] Browser exploration against isolated local Demo origins; focused follow-ups verified application Refresh/Reload and restoration

No product source changes. Hardware/flash guarantees, authenticated Abyss writes, exhaustive boundaries and narrow-screen coverage remain explicitly unverified. The current Demo does not provide Devtool. A malformed Release Notes hash causing an empty page was reproduced and recorded as an unaccepted, unfixed candidate, not silently normalized into the specification.

Start at [docs/spec/README.md](docs/spec/README.md). Historical trial reports and their review corrections can be inspected at [dc4ac4a](https://github.com/cormoran/dya-studio/tree/dc4ac4a/docs/spec/validation), but are removed from HEAD; this is not a policy to commit every future trial. Squash merging may omit intermediate commits from main history.

Observed scope: keymap modal/floating behavior and Save/Reload; Home language switching; Connection cancel/default-layer restoration; Settings debounce/Reload and all-source Demo restoration; Subsystems disable/reconnect/unavailable/restore; Macro Save/Refresh; Trackball scaling/axis-snap Reload; diagnostic expansion/copy feedback; Guide navigation/fallback; Release Notes list/hash/language; OAuth configuration-error recovery; and Import/Export signed-out gating. Initial readback/restoration overclaims were corrected in focused follow-ups. Full hardware, authentication, failure injection, rotary editing, history restore, capacity boundaries and narrow-view coverage were not established. The malformed hash reproduction remains in the Release Notes specification with its observed conditions.

## Product specifications

- [x] Added page/module specifications and maintenance rules; requirements inferred from code are explicitly labeled
- [x] Ran `npm run spec:check`

## Release notes

- [x] Internal documentation/tooling only; no user-visible application behavior change

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
+## Product specifications
+
+<!-- Follow docs/spec/AUTHORING.md. Name affected specification IDs and update
+their requirements/current behavior/flows/errors in the same PR. If no observable
+behavior changed, identify the reviewed specifications and explain why no update
+is needed. Browser reports should separate evidence from blocked/not-run cases. -->
+
+- [ ] Updated affected specifications, or explained why no update is needed
+- [ ] Ran `npm run spec:check` when changing specifications
+
```
Comment (cormoran) [inline .github/pull_request_template.md]: I think it's too much. Let's remove this check list.
